pkg/sql/executor.go

```diff
@@ -763,12 +791,11 @@ func (e *Executor) execParsed(
 			}
 
 			var err error
-			if results != nil {
-				// Some results were produced by a previous attempt. Discard them.
-				ResultList(results).Close(ctx)
+			if !resultWriter.CanRetry() {
```

Comment:
This will be changed to do what's described in the RFC (e.g. set the transaction state from `FirstBatch` to `Open` rather than checking on each attempt to run the `txnClosure`)